odoo/addons/base/models/res_company.py

```diff
@@ -290,3 +345,19 @@ def _get_main_company(self):
             main_company = self.env['res.company'].sudo().search([], limit=1, order="id")
 
         return main_company
+
+    @ormcache('tuple(self.env.companies.ids)', 'self.id')
+    def __accessible_branches(self):
+        self.ensure_one()
+        children = self.sudo().search([('id', 'child_of', self.id)])
+        accessible = self.env.companies
+        return (children & accessible).ids
+
+    def _accessible_branches(self):
+        return self.browse(self.__accessible_branches())
+
+    def _all_branches_selected(self):
+        return (
+            len(self.env.companies.root_id) == 1
+            and self.env.companies == self.env['res.company'].sudo().search([('id', 'child_of', self.env.company.root_id.ids)])
```

Comment:
Because this check is general and doesn't depend on your rights regarding the main company/branches

It is used to disable some actions that only make sense if you have access (and access is activated) to all branches of a company